app/editor/src/features/admin/sources/SourceDetails.tsx

```diff
@@ -19,6 +19,7 @@ export const SourceDetails: React.FC<ISourceDetailsProps> = () => {
 
   const users = getUserOptions(lookups.users);
   const licenses = getSortableOptions(lookups.licenses);
+  const products = getSortableOptions(lookups.products);
```

Comment:
There is an override argument that allows you to prepend options.  You can prepend an option that handles `undefined` value.